lib/utils.js

```diff
@@ -544,6 +544,17 @@ exports.getError = function (err) {
   return err || exports.undefinedError();
 };
 
+/**
+ * Show a deprecation warning
+ *
+ * @param {string} msg
+ */
+exports.deprecate = process.emitWarning
```

Comment:
Sorry to complicate this, but it's a bit more complicated.  😝 

1.  [ ] The user should only see this notice *once*.  For each unique `msg`, we need some to keep some flag (probably in an object, as its key, since we have no access to `Set`) and do nothing if the flag is already set.  For safety, coerce `msg` into a `string` via `msg = String(msg)`
2.  [ ] Internally, `process.emitWarning` emits on the next tick.  So, `console.warn(msg)` must happen within `process.nextTick` as well.  This is to avoid bloating or interfering w/ the call stack

Reply:
Also, please rebase onto `master` to make the Snyk warning go away

Reply:
If `process.nextTick` is unavailable (outside of Node), is it preferred to use `Promise.resolve().then(…)` or `setTimeout(…, 0)`?

Reply:
Does this also need to hook into the `--no-deprecation` flags?